### DISCUSSION TITLE
Using in custom framework - cant use dynamic require/import

### DISCUSSION THREAD

**Discussion by MrEssex** [inertiajs/inertia#1072] [Help (Svelte)] [answered]
Hi there, I'm using Inertia in a custom Framework. 

Due to this and the way we require resources, dynamic imports/require aren't an option. 
I'm not 100% certain when it comes to js and assumed i could just require the components and try to resolve it with a switch statement, but I cant get this working. 

Nothing too complex just trying to reach lift off at this stage with some basic tests. 

```
import {createInertiaApp} from '@inertiajs/inertia-svelte';

import Home from './Pages/Home.svelte';
import About from './Pages/About.svelte';
import Contact from './Pages/Contact.svelte';

export default createInertiaApp({
  resolve: name => resolver(name),
  setup({el, App, props})
  {
    new App({target: el, props});
  }
});

function resolver(name)
{
  switch(name)
  {
    case 'home':
      return Home.default;
    case 'about':
      return About.default;
    case 'contact':
      return Contact.default;
    default:
      return Home.default;
  }
}
```
What I'm wondering is, has anyone else encountered this situation and how did you go about resolving the issue? / Can anybody point me in the right direction?

**Comment by ajnsn** [comment, marked as answer]
Hey @MrEssex 

> Due to this and the way we require resources, dynamic imports/require aren't an option.

I might not fully understand what you mean with that but CommonJS`require()` at it's core should include all components in your bundle and isn't dynamic? Which bundler are you using?

**Reply by MrEssex** [reply]
Apologies @ajnsn, it was an issue with the build process using Rollup. 

Turns out I could just require the `inlineDynamicImports` flag which fixes my issue. 
Thank you
